### PR TITLE
REFPLTB-1457: Building wifi hal code with WIFI_HAL_VERSION_3 for Turr…

### PR DESCRIPTION
**Pull request by manigandanj**
Signed-off-by: Manigandan Gopalakrishnan <manigandan.gopalakrishnan@ltts.com>

### CODE REVIEW

File: source/wifi/wifi_hal.c [hunk 1]
```diff
@@ -8839,28 +9012,26 @@ INT wifi_createVAP(wifi_radio_index_t index, wifi_vap_info_map_t *map)
...
 
     /* get the first token */
-    token = strtok(pchannels, s);
+    token = strtok_r(pchannels, s, &next);
 
     /* walk through other tokens */
-    while( token != NULL ) {
-        chlistptr->channels_list[count] = atoi(token);
-        count++;
-        token = strtok(NULL, s);
```
Comment (manigandanj) [inline source/wifi/wifi_hal.c:8856]: i think, this will elicit SIGSEGV.

File: source/wifi/wifi_hal.c [hunk 2]
```diff
@@ -1517,35 +1514,28 @@ INT wifi_setRadioChannelMode(INT radioIndex, CHAR *channelMode, BOOL gOnlyFlag,
...
-    char cmd[MAX_CMD_SIZE] = {0};
-    int count = 0;
-    if (NULL == output_string)
+    //Getting phy index corresponding to radio index
+    sprintf(cmd, "iw dev wlan0 info | grep wiphy | xargs | cut -d ' ' -f2", radioIndex);
+    ret = _syscmd(cmd, output, sizeof(output));
+    if ((ret != 0) || (strlen(output) == 0))
         return RETURN_ERR;
+    phyIndex = atoi(output);
+    printf("MANIII: phyIndex: %d\n", phyIndex);
```
Comment (mbajorski) [inline source/wifi/wifi_hal.c:1528]: Can we remove MANIII debug prints or use wifi_dbg_printf

File: source/wifi/wifi_hal.c [hunk 3]
```diff
@@ -9013,7 +9043,90 @@ INT wifi_getRadioOperatingParameters(wifi_radio_index_t index, wifi_radio_operat
...
+    if(wifi_getApWpsDevicePIN(apIndex,&pin) != RETURN_OK)
+    {
+        printf("wifi_getApWpsDevicePIN failed for apIndex: %d\n", apIndex);
+        return RETURN_ERR;
+    }
+    sprintf(wpsConfig->pin,"%ld",pin);
+    /* Get enabled WPS config methods */
+    if(wifi_getApWpsConfigMethodsEnabled(apIndex,wps_enabled_methods) != RETURN_OK)
+    {
+        printf("wifi_getApWpsConfigMethodsEnabled failed for apIndex: %d\n", apIndex);
```
Comment (mbajorski) [inline source/wifi/wifi_hal.c:9070]: wifi_dbg_printf

File: source/wifi/wifi_hal.c [hunk 4]
```diff
@@ -9025,11 +9138,11 @@ INT wifi_getRadioVapInfoMap(wifi_radio_index_t index, wifi_vap_info_map_t *map)
     map->num_vaps = 3; // XXX: this is a hack. For both radio let's support 3 vaps for now
                        //      home, backhaul and onboard.
     WIFI_ENTRY_EXIT_DEBUG("Inside %s:%d\n",__func__, __LINE__);
-    printf("Entering %s index = %d", __func__, (int)index);
+    printf("Entering %s radioIndex = %u", __func__, radioIndex);
```
Comment (mbajorski) [inline source/wifi/wifi_hal.c:9141]: I'm aware that this is just log content change but can we also change printf to wifi_dbg_printf?

**Comment by scthunderbolt** [comment]
Closed due to inactivity, please restore if still needed.

**Pull request closed without merging**